crates/precompiles/src/tip20/mod.rs

```diff
@@ -845,13 +720,8 @@ impl TIP20Token {
     }
 
     /// Checks and updates spending limits for access keys.
-    /// Only active after Allegro Moderato hardfork.
     pub fn check_spending_limit(&mut self, from: Address, amount: U256) -> Result<()> {
```

Comment:
nit: we should consider updating this to `check_and_update_spending_limit` or similar since this does both. The fn name might be confusing to some indicating that its just checking the limit.